Tests/ClientTests/Mocks/MockTests/MockSyncManagerTests.swift

```diff
@@ -0,0 +1,65 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+import Foundation
+
+@testable import Client
+
+final class MockSyncManagerTests: XCTestCase {
```

Comment:
This looks good, just a comment on the naming conventions as they are little confusing/misleading.

Can you rename this to ClientSyncManagerTests as that is ultimately what you are trying to test here (correct me if I have misinterpreted something)

Can you also rename the MockSyncManager to ClientSyncManagerSpy as I think it would clarify what the test set up is. Mock implies it's just a fake version of something which isn't the case here.